src/core/render/emojify.js

```diff
@@ -1,20 +1,45 @@
-import { inBrowser } from '../util/env';
+import emojiData from './emojify-data.js';
 
-function replace(m, $1) {
-  return (
-    '<img class="emoji" src="https://github.githubassets.com/images/icons/emoji/' +
-    $1 +
-    '.png" alt="' +
-    $1 +
-    '" />'
-  );
+function replaceEmojiShorthand(m, $1, useNativeEmoji) {
+  const emojiMatch = emojiData.data[$1];
+
+  let result = m;
+
+  if (emojiMatch) {
+    if (useNativeEmoji && /unicode/.test(emojiMatch)) {
+      const emojiUnicode = emojiMatch
+        .replace('unicode/', '')
+        .replace(/\.png.*/, '')
+        .split('-')
+        .map(u => `&#x${u};`)
+        // Separate multi-character emoji with zero width joiner sequence (ZWJ)
+        // Hat tip: https://about.gitlab.com/blog/2018/05/30/journey-in-native-unicode-emoji/#emoji-made-up-of-multiple-characters
+        .join('&zwj;')
+        .concat('&#xFE0E;');
+      result = `<span class="emoji">${emojiUnicode}</span>`;
+    } else {
+      result = `<img src="${emojiData.baseURL}${emojiMatch}.png" alt="${$1}" class="emoji" loading="lazy">`;
+    }
+  }
+
+  return result;
 }
 
-export function emojify(text) {
-  return text
-    .replace(/<(pre|template|code)[^>]*?>[\s\S]+?<\/(pre|template|code)>/g, m =>
-      m.replace(/:/g, '__colon__')
-    )
-    .replace(/:([a-z0-9_\-+]+?):/g, (inBrowser && window.emojify) || replace)
-    .replace(/__colon__/g, ':');
+export function emojify(text, useNativeEmoji) {
+  return (
+    text
+      // Mark colons in tags
+      .replace(
+        /<(code|pre|script|template)[^>]*?>[\s\S]+?<\/(code|pre|script|template)>/g,
+        m => m.replace(/:/g, '__colon__')
+      )
+      // Mark colons in comments
+      .replace(/<!--[\s\S]+?-->/g, m => m.replace(/:/g, '__colon__'))
+      // Replace emoji shorthand codes
+      .replace(/:([\w\-+]+?):/g, (m, $1) =>
```

Comment:
About '\w': the idea between 'a-z0-9_' was to reduce false-positives, a word being 'a-zA-Z0-9_' and github emojis being lowercase.

Reply:
> 

Actually, I think we support the supper case in preview versions as well.

Reply:
@ChoKaPeek: Good point. Fixed.

@Koooooo-7: Docsify's emoji parser wasn't case-sensitive but GitHub's servers are. That meant that Docsify would treat `:smile:` and `:SMILE:` the same but only the first would render properly. It therefore makes sense to restore @ChoKaPeek's RegEx change.

Example:
* https://github.githubassets.com/images/icons/emoji/smile.png
* https://github.githubassets.com/images/icons/emoji/SMILE.png